### PR TITLE
UCP: Enhance the requirement when selecting sockaddr aux transports.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6997/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe Can you please take a look?

File: src/ucp/wireup/wireup_ep.c [hunk 1]
```diff
@@ -429,10 +429,16 @@ static ucs_status_t ucp_wireup_ep_pack_sockaddr_aux_tls(ucp_worker_h worker,
...
+     * It also has to be a UCT_IFACE_FLAG_CONNECT_TO_IFACE transport and support
+     * active messaging */
     ucs_for_each_bit(tl_id, context->config.sockaddr_aux_rscs_bitmap) {
-        if (!strncmp(context->tl_rscs[tl_id].tl_rsc.dev_name, dev_name,
-                     UCT_DEVICE_NAME_MAX)) {
+        if ((!strncmp(context->tl_rscs[tl_id].tl_rsc.dev_name, dev_name,
+                      UCT_DEVICE_NAME_MAX)) &&
+            (worker->ifaces[tl_id].attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) &&
+            (worker->ifaces[tl_id].attr.cap.flags & (UCT_IFACE_FLAG_AM_SHORT |
+                                                     UCT_IFACE_FLAG_AM_BCOPY |
```
Comment (yosefe) [inline src/ucp/wireup/wireup_ep.c]: need to require explicitly AM_BCOPY since wireup_msg uses am_bcopy

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7010/ for details (Mellanox internal link).

**Pull request merged**